controllers/chat/initChatFromPost.js

```diff
@@ -16,13 +16,16 @@ const get_anon_info_from_comment_post = async (client, userModel, source_id) =>
   });
   let comments = post_detail?.own_reactions?.comment;
   // iterate comments
-  for (const element of comments) {
-    let comment = element;
-    if (comment.user_id === userModel.user_id) {
-      comment_data = comment;
-      break;
+  if (comments) {
+    for (const element of comments) {
+      let comment = element;
+      if (comment.user_id === userModel.user_id) {
+        comment_data = comment;
+        break;
+      }
     }
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `if (comments)` check is a good improvement to handle cases where there are no comments. However, it would be better to also check if `comments` is an array before iterating over it. This will prevent potential errors if `comments` is not null or undefined but also not an array.
```diff
-   if (comments) {
+   if (Array.isArray(comments)) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->